src/providers/oneNoteApiDataProvider.ts

```diff
@@ -141,58 +130,32 @@ export class OneNoteApiDataProvider implements OneNoteDataProvider {
 	}
 
 	getSpNotebookProperties(spNotebook: SharedNotebook, expands?: number, excludeReadOnlyNotebooks?: boolean): Promise<SharedNotebookApiProperties> {
-		return new Promise<SharedNotebookApiProperties>((resolve, reject) => {
-			this.getNotebookSelfUrlFromSpUrl(spNotebook.webUrl).then((selfUrl) => {
-				this.http('GET', selfUrl + '?' + this.getExpands(expands), this.authHeader, this.headers).then((xhr) => {
-					const notebook: OneNoteApi.Notebook = xhr.response && JSON.parse(xhr.response);
-					if (notebook) {
-						spNotebook.apiUrl = notebook.self;
-						const spSections = notebook.sections.map(section => this.responseTransformer.transformSection(section, spNotebook));
-						const spSectionGroups = notebook.sectionGroups.map(sectionGroup => this.responseTransformer.transformSectionGroup(sectionGroup, spNotebook));
-						resolve({
-							id: notebook.id,
-							spSectionGroups: spSectionGroups,
-							spSections: spSections
-						});
-						return;
-					}
-					reject(xhr);
-				}).catch((xhr) => {
-					reject(xhr);
-				});
-			}).catch((xhr) => {
-				reject(xhr);
+		return this.getNotebookSelfUrlFromSpUrl(spNotebook.webUrl).then((selfUrl) => {
+			return this.api.performApiCall(selfUrl + '?' + this.getExpands(expands), undefined /* data */, undefined /* content type */, 'GET' /* http method */, true /* isFullUrl */).then(response => {
+				const notebook: OneNoteApi.Notebook = response.parsedResponse;
+				if (notebook) {
+					spNotebook.apiUrl = notebook.self;
+					const spSections = notebook.sections.map(section => this.responseTransformer.transformSection(section, spNotebook));
+					const spSectionGroups = notebook.sectionGroups.map(sectionGroup => this.responseTransformer.transformSectionGroup(sectionGroup, spNotebook));
+					return {
+						id: notebook.id,
+						spSectionGroups: spSectionGroups,
+						spSections: spSections
+					};
+				}
+				throw new Error('could not find notebook in get notebooks expands request');
```

Comment:
Minor, but it may be good to keep the error messages consistent. This one starts with a lower case letter and has no punctuation, while the other message starts with a capital letter and ends with an exclamation point.

Reply:
Fixed now